diff --git a/CMakeLists.txt b/CMakeLists.txt
index 25ad376..3bd5e5d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -844,7 +844,7 @@ CHECK_CXX_SOURCE_COMPILES(
 )
 
 IF(NOT VMIME_SHARED_PTR_USE_CXX AND NOT VMIME_SHARED_PTR_USE_BOOST)
-	IF(CXX11_COMPILER_FLAGS AND VMIME_HAS_CXX11_SHARED_PTR)
+	IF(VMIME_HAS_CXX11_SHARED_PTR)
 		# If std::shared_ptr<> is available, use it by default
 		SET(VMIME_SHARED_PTR_USE_CXX_DEFAULT     ON)
 		SET(VMIME_SHARED_PTR_USE_BOOST_DEFAULT  OFF)
@@ -964,6 +964,23 @@ IF(WIN32)
 
 ENDIF()
 
+if(MSVC)
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+  add_definitions(-DU_STATIC_IMPLEMENTATION)
+    foreach(flag_var
+            CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
+            CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO
+            CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+            CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+      if(${flag_var} MATCHES "/MD")
+        string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+      endif()
+      if(${flag_var} MATCHES "/MDd")
+        string(REGEX REPLACE "/MDd" "/MTd" ${flag_var} "${${flag_var}}")
+      endif()
+    endforeach(flag_var)
+endif()
+
 
 ##############################################################################
 # POSIX-specific checks
diff --git a/cmake/FindICU.cmake b/cmake/FindICU.cmake
index 786f157..eacd07c 100644
--- a/cmake/FindICU.cmake
+++ b/cmake/FindICU.cmake
@@ -90,12 +90,12 @@ macro(icu_declare_component _NAME)
     set("${ICU_PRIVATE_VAR_NS}_COMPONENTS_${_NAME}" ${ARGN})
 endmacro(icu_declare_component)
 
-icu_declare_component(data icudata)
-icu_declare_component(uc   icuuc)         # Common and Data libraries
-icu_declare_component(i18n icui18n icuin) # Internationalization library
-icu_declare_component(io   icuio ustdio)  # Stream and I/O Library
-icu_declare_component(le   icule)         # Layout library
-icu_declare_component(lx   iculx)         # Paragraph Layout library
+icu_declare_component(data sicudt)
+icu_declare_component(uc   sicuuc)         # Common and Data libraries
+icu_declare_component(i18n sicui18n sicuin) # Internationalization library
+icu_declare_component(io   sicuio sustdio)  # Stream and I/O Library
+icu_declare_component(le   sicule)         # Layout library
+icu_declare_component(lx   siculx)         # Paragraph Layout library
 
 ########## Public ##########
 set(${ICU_PUBLIC_VAR_NS}_FOUND TRUE)
